### ISSUE TITLE
Error when using http/https scheme with UriSource

### ISSUE THREAD

**Issue by BrandonHaynes** [uwescience/myria#721]
Consider the plan below, which sources data from a single URL.  

Upon execution, Myria reports `java.io.IOException: No FileSystem for scheme: http` (full exception below).  I'm virtually positive that I have successfully loaded data using an http scheme (the only other possibility is that I was using a custom `DataSource` implementation).  It's also possible that I had frameworks that expose a `HttpFileSystem` co-installed at the time (e.g., Hadoop's HttpFS).

Is this a regression bug, a configuration issue, or does the UriSource really not currently support these schemes?  It's also possible that I've gone insane, because I was sure that this used to work.

(Originally observed by @ryanmaas) 

```json
{
   "fragments":[
      {
         "workers":[
            1
         ],
         "operators":[
            {
               "source":{
                  "dataType":"URI",
                  "uri":"http://s3-us-west-2.amazonaws.com/myria-sdss/oceanography/smallsample_ocean_seq.csv"
               },
               "opId":0,
               "opType":"FileScan",
               "schema":{
                  "columnNames":[
                     "col",
                     "col2",
                     "col3",
                     "col4",
                     "col5"
                  ],
                  "columnTypes":[
                     "LONG_TYPE",
                     "DOUBLE_TYPE",
                     "DOUBLE_TYPE",
                     "DOUBLE_TYPE",
                     "DOUBLE_TYPE"
                  ]
               }
            },
            {
               "argOverwriteTable":true,
               "opId":1,
               "opType":"DbInsert",
               "relationKey":{
                  "userName":"public",
                  "relationName":"testMyriaUpload",
                  "programName":"adhoc"
               },
               "argChild":0
            }
         ]
      }
   ],
   "logicalRa":"Parallel Import [(1, 'http://s3-us-west-2.amazonaws.com/myria-sdss/oceanography/smallsample_ocean_seq.csv')]",
   "rawQuery":"Parallel Import [(1, 'http://s3-us-west-2.amazonaws.com/myria-sdss/oceanography/smallsample_ocean_seq.csv')]"
}
```

Full exception:

```
edu.washington.escience.myria.DbException: Query #66690.0 failed: java.io.IOException: No FileSystem for scheme: http\n\tat edu.washington.escience.myria.parallel.MasterSubQuery$WorkerExecutionInfo$2.operationComplete(MasterSubQuery.java:105)
	at edu.washington.escience.myria.parallel.LocalSubQueryFutureListener.operationComplete(LocalSubQueryFutureListener.java:45)
	at edu.washington.escience.myria.util.concurrent.OperationFutureBase.notifyListener(OperationFutureBase.java:583)
	at edu.washington.escience.myria.util.concurrent.OperationFutureBase.notifyListeners(OperationFutureBase.java:542)
	at edu.washington.escience.myria.util.concurrent.OperationFutureBase.wakeupWaitersAndNotifyListeners(OperationFutureBase.java:151)
	at edu.washington.escience.myria.util.concurrent.OperationFutureBase.setFailure0(OperationFutureBase.java:506)
	at edu.washington.escience.myria.parallel.LocalSubQueryFuture.setFailure(LocalSubQueryFuture.java:69)
	at edu.washington.escience.myria.parallel.MasterSubQuery.workerFail(MasterSubQuery.java:320)
	at edu.washington.escience.myria.parallel.QueryManager.workerFailed(QueryManager.java:308)
	at edu.washington.escience.myria.parallel.Server$MessageProcessor.run(Server.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at edu.washington.escience.myria.util.concurrent.RenamingThreadFactory$1.run(RenamingThreadFactory.java:33)
	Suppressed: edu.washington.escience.myria.DbException: Worker #1 failed: java.io.IOException: No FileSystem for scheme: http
		at edu.washington.escience.myria.operator.FileScan.init(FileScan.java:249)
		at edu.washington.escience.myria.operator.Operator.open(Operator.java:402)
		at edu.washington.escience.myria.operator.Operator.open(Operator.java:392)
		at edu.washington.escience.myria.parallel.LocalFragment.initActually(LocalFragment.java:707)
		at edu.washington.escience.myria.parallel.LocalFragment.access$700(LocalFragment.java:44)
		at edu.washington.escience.myria.parallel.LocalFragment$2.call(LocalFragment.java:225)
		at edu.washington.escience.myria.parallel.LocalFragment$2.call(LocalFragment.java:218)
		at edu.washington.escience.myria.util.concurrent.ExecutableExecutionFuture.call(ExecutableExecutionFuture.java:193)
		... 4 more
	Caused by: edu.washington.escience.myria.DbException: java.io.IOException: No FileSystem for scheme: http
		... 12 more
	Caused by: java.io.IOException: No FileSystem for scheme: http
		at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:1443)
		at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:67)
		at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:1464)
		at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:263)
		at edu.washington.escience.myria.io.UriSource.getInputStream(UriSource.java:56)
		at edu.washington.escience.myria.operator.FileScan.init(FileScan.java:241)
		... 11 more
```

**Comment by dhalperi** [comment]
Good timing -- I just sent a mail to the Myria list that has the below content.

I suspect somewhere along the line the hadoop jar loading got broken. Andrew definitely tested s3, http, and file links when he wrote this code. (Although we clearly need unit tests of them!)

Relevant stack overflow issue:
http://stackoverflow.com/questions/17265002/hadoop-no-filesystem-for-scheme-file

**Comment by BrandonHaynes** [comment]
All of the SO links that appeared relevant deal with the file scheme, which does works (as does hdfs).  So this isn't confusion between the FileSystem and DistributedFileSystem locations.  What boggles me is that it doesn't appear that hadoop-core-1.2.1.jar _ever_ supported http (though I haven't looked very deeply here).

I did clone a commit from approximately August of 2014 and verified that the problem exists as far back as then.

**Comment by dhalperi** [comment]
@BrandonHaynes I think we're both crazy, is the answer.

**Comment by BrandonHaynes** [comment]
How would you feel about explicitly checking for http/https schemes and just getting a stream via `URL.openConnection().getInputStream()`?  I suspect that, overall, this might be cleaner than registering a HTTP-oriented FileSystem implementation.

**Comment by dhalperi** [comment]
Seems reasonable to me.

**Comment by dhalperi** [comment]
@BrandonHaynes FYI, I think if you had used "Fix #721" instead of "Corrects #721" in the PR description for #727 then GitHub would have auto-closed this upon merging :)
